mapsforge-map-android/src/main/java/org/mapsforge/map/android/graphics/AndroidPaint.java

```diff
@@ -146,11 +151,118 @@ public int getTextHeight(String text) {
         return rect.height();
     }
 
+    @Override
+    public int getTextHeight(final String text, boolean includePadding) {
+        int retVal = getTextHeight(text);
+
+        if (includePadding && retVal > 0) {
+            retVal += getFontPadding();
+        }
+
+        return retVal;
+    }
+
     @Override
     public int getTextWidth(String text) {
         return (int) this.paint.measureText(text);
     }
 
+    @Override
+    public int getTextWidth(String text, boolean includePadding) {
+        int retVal = getTextWidth(text);
+
+        if (includePadding && retVal > 0) {
+            retVal += getFontPadding();
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Uses StaticLayout to measure text accurately, and respects Spans.
+     * Returns width of the widest line.
+     */
+    @SuppressWarnings("deprecation")
+    @Override
+    public int getTextWidth(final String text, final int widthMax, boolean includePadding) {
+        int retVal = 0;
+
+        if (text != null && widthMax >= 0) {
```

Comment:
@Sublimis Why `widthMax >= 0` and not `widthMax > 0`?

Reply:
This snippet is taken from our app so not sure any more, however by looking at the StaticLayout, width parameter value of zero is valid, so that is probably the reason. Although there's no explanation anywhere in the StaticLayout documentation why width=0 is accepted. But it is.

Note that these three-argument `getTextWidth(, ,)` methods are not currently used anywhere, but thought they might be useful in the future (the AndroidPaint one, that is). We used it a little while developing this PR, until decided on a different approach. They can be safely deleted for now (from the Paint interface too), you decide.

Reply:
> Note that these three-argument getTextWidth(, ,) methods are not currently used anywhere
> They can be safely deleted for now (from the Paint interface too), you decide.

If they are not used, it is better to remove them, as the public API is difficult to remove in the future.

And we avoid for now the desktop implementation. :slightly_smiling_face:

Reply:
@Sublimis 

> Note that these three-argument getTextWidth(, ,) methods are not currently used anywhere
> They can be safely deleted for now (from the Paint interface too), you decide.

If they are not used, it is better to remove them, as the public API is difficult to remove in the future.

And we avoid for now the desktop implementation. :slightly_smiling_face:

Reply:
Very true!